Tests/HarmonyTests/CacheRepositoryTests.swift

```diff
@@ -0,0 +1,213 @@
+//
+// Copyright 2022 Mobile Jazz SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import Nimble
+import XCTest
+import Harmony
+import HarmonyTesting
+
+class CacheRepositoryTests: XCTestCase {
+    
+    private var main: InMemoryDataSource<Int>!
+    private var cache: InMemoryDataSource<Int>!
+    
+    private typealias DataSourceSpyType = DataSourceSpy<InMemoryDataSource<Int>, Int>
+    
+    private var mainSpy: DataSourceSpyType!
+    private var cacheSpy: DataSourceSpyType!
+    
+    private var cacheRepository: CacheRepository<DataSourceSpyType, DataSourceSpyType, Int>!
+
+    private func setup(
+        mainData mainKeyValue: (query: KeyQuery, value: Int)?,
+        cacheData cacheKeyValue: (query: KeyQuery, value: Int)?,
+        objectValid: Bool = true
+    ) throws {
+        main = InMemoryDataSource()
+        cache = InMemoryDataSource()
+        
+        mainSpy = DataSourceSpy(main)
+        cacheSpy = DataSourceSpy(cache)
+        
+        cacheRepository = CacheRepository(
+            main: mainSpy,
+            cache: cacheSpy,
+            validator: MockObjectValidation<Int>(objectValid: objectValid, arrayValid: true)
+        )
+        
+        if let mainKeyValue = mainKeyValue {
+            _ = try main.put(mainKeyValue.value, in: mainKeyValue.query).result.get()
+        }
+        
+        if let cacheKeyValue = cacheKeyValue {
+            _ = try cache.put(cacheKeyValue.value, in: cacheKeyValue.query).result.get()
+        }
+    }
+    
+    func test_getValue_withMain_withEmptyMain() throws {
+        // Given
+        let query = IdQuery("id")
+        try setup(mainData: nil, cacheData: nil)
+        let operation = MainOperation()
+        
+        // When
+        do {
+            _ = try cacheRepository.get(query, operation: operation).result.get()
+            XCTAssert(false)
+        } catch {
+            // Then
+            expect(error).to(beAnInstanceOf(CoreError.NotFound.self))
+        }
```

Comment:
Could be better to follow another style to check the error, see the in the `InMemoryDataSourceTests`.